src/game_api/script.cpp

```diff
@@ -1037,6 +1037,29 @@ SpelunkyScript::ScriptImpl::ScriptImpl(std::string script, std::string file, Sou
         }
         return (uint16_t)0;
     };
+    /// <summary>
+    /// Read input that has been previously stolen with steal_input
+    /// </summary>
+    lua["read_stolen_input"] = [this](int uid)
+    {
+        Player* player = get_entity_ptr(uid)->as<Player>();
+        if (player == nullptr)
+            return (uint16_t)0;
+        ScriptInput* readinput = reinterpret_cast<ScriptInput*>(player->input_ptr);
+        if (!IsBadReadPtr(readinput, 20))
+        {
+            readinput = reinterpret_cast<ScriptInput*>(readinput->orig_input);
+            if (!IsBadReadPtr(readinput, 20))
+            {
+                // idk what 7712 is but it seems like it's an uninitialized default value that we should ignore
+                if (readinput->next != 7712)
```

Comment:
Is this value perhaps just the default buttons? Say if you were to not have run by default, does this value look different?

Reply:
If I have no buttons pressed the value is 0. I wonder if this is an and of all the buttons?

Reply:
looks like it was the <summary> tag that vs added. I removed it and regenerated the file

Reply:
It's `00011110 00100000` which we could translate into what's pressed and not but off the top of my head I can't say  😅

Reply:
It's:
- Door pressed
- Whip released
- Bomb released
- Rope released
- Run released

Don't think that makes much sense

Reply:
this is now unneeded by checking for the stolen input